frontend/model/contracts/shared/time.js

```diff
@@ -257,3 +259,39 @@ export function cycleAtDate (atDate: string | Date): number {
   const partialCycles = now.getDate() / lastDayOfMonth(now).getDate()
   return partialCycles
 }
+
+export function timeLeft (expiryTime: number) {
+  const now = new Date()
+  const expiry = new Date(expiryTime)
+
+  if (expiry < now) {
+    return {}
+  }
```

Comment:
The `expired` looks like a good idea, but I'm not sure about setting the other fields to nonsensical values (nothing other than `null`, `undefined` or `NaN` makes sense, IMO). With TS (and likely Flow too) you can tell the two shapes apart `{ expired: false, year: number, ... } | { expired: false }` and at worst rely on type guard functions.

Reply:
The other values that would make sense would be (negative?) numbers indicating how long ago expiry happened, but this sounds like an unneeded complication since we don't use those values.

Reply:
Yeah, but I don't get the point of defining a value that will not be used. In fact, the reason for returning an empty object was to simplify code (to support destructuring and to avoid needing a try block), but setting `year` and all other fields to 0 (or any other value, except negative numbers that say when it expired) would provide exactly nil useful information

Reply:
Because it is useful to developers who are using this API. Say you are working on the frontend and want to display the amount of time left on an invite. You can use these values and bind them to the UI, to show the user that the invite has "0 seconds left", and the color it in red. Or say you want to destructure the return value using the same logic and avoid having an if/else based on whether or not the `years` value is defined. Or say you want to, in the future, using TypeScript, define a custom type that represents the return value of this function. Again, that would be simpler by keeping the same object shape.

Reply:
> Or say you want to, in the future, using TypeScript, define a custom type that represents the return value of this function

Like mentioned, that's not an issue at all. I even provided an example. In fact, I think it's encouraged in TypeScript to use objects with different shapes rather than try to fit everything into a one-size-fits-all type.

> You can use these values and bind them to the UI, to show the user that the invite has "0 seconds left"

We don't return seconds, but even then you shouldn't be displaying '0' seconds, you should be checking if `expired` is true or not.

In fact, the logic would be more complicated because you'd need something like:

```
if (years === 0 && months === 0 && hours === 0 && minutes === 0) {
return '0 but in red'
} else {
return `{years}:{months}:{hours}:{minutes} left`
}
```

as opposed to 

```
if (expired) {
return '0 but in red'
} else {
return `{years}:{months}:{hours}:{minutes} left`
}
```

> Or say you want to destructure the return value using the same logic and avoid having an if/else based on whether or not the years value is defined

Sure, but you shouldn't be doing this to begin with (checking if `year` is defined).

Reply:
> In fact, the logic would be more complicated because you'd need something like:

That's not what I had in mind... rather I was thinking something like this:

```pug
i18n(:class="{ red: expired }" :args="{ minutes }") {minutes} left
```

I don't see a downside to returning `return { expired: true, years: 0, months: 0, days: 0, hours: 0, minutes: 0 }`, but I do see potential negative unintended consequences of not doing so.